server/src/services/initialization/featureFlags.ts

```diff
@@ -0,0 +1,76 @@
+import got from "got";
+import crypto from "crypto";
+import { ServerState } from "../../types";
+import { isTestMode } from "../../utils";
+
+export interface FeatureFlag {
+  percent: number;
+}
+
+export interface FeatureFlags {
+  semanticHighlighting: FeatureFlag;
+  documentSymbol: FeatureFlag;
+}
+
+const DEFAULT_FLAGS: FeatureFlags = {
+  semanticHighlighting: {
+    percent: 0,
+  },
+  documentSymbol: {
+    percent: 0,
+  },
+};
+
+export async function fetchFeatureFlags(
+  state: ServerState
+): Promise<FeatureFlags> {
+  state.logger.info("Fetching feature flags");
+
+  try {
+    return await got
+      .get(
+        "https://raw.githubusercontent.com/antico5/hardhat-vscode/flags/flags.json",
```

Comment:
I suggest including that file in this PR, and changing the URL to:
Developers can always edit the URL locally for testing.

```suggestion
        "https://raw.githubusercontent.com/NomicFoundation/hardhat-vscode/development/flags.json",
```

Reply:
I learned about `ServerState.env` today. Maybe we can use it here?

- If `state.env == "production"` we can use that URL.
- If `state.env == "development"` we can load that file from the local repo.

Just a suggestion.

Reply:
When on development mode, we have features enabled, we do that on the `isFeatureEnabled` function.